framework/src/materials/VectorMagnitudeFunctorMaterial.C

```diff
@@ -20,38 +20,60 @@ VectorMagnitudeFunctorMaterialTempl<is_ad>::validParams()
   InputParameters params = FunctorMaterial::validParams();
   params += SetupInterface::validParams();
   params.set<ExecFlagEnum>("execute_on") = {EXEC_ALWAYS};
-  params.addClassDescription("This class takes up to three functors corresponding to vector "
-                             "components and computes the Euclidean norm from them.");
-  params.addRequiredParam<MooseFunctorName>("x_functor",
-                                            "The functor corresponding to the x component.");
+  params.addClassDescription(
+      "This class takes up to three scalar-valued functors corresponding to vector "
+      "components *or* a single vector functor and computes the Euclidean norm.");
+  params.addParam<MooseFunctorName>("x_functor", "The functor corresponding to the x component.");
   params.addParam<MooseFunctorName>(
       "y_functor", 0, "The functor corresponding to the y component.");
   params.addParam<MooseFunctorName>(
       "z_functor", 0, "The functor corresponding to the z component.");
-  params.addRequiredParam<MooseFunctorName>("vector_functor_name",
-                                            "The name of the vector functor that we are creating.");
+  params.addRequiredParam<MooseFunctorName>(
+      "vector_magnitude_name", "The name of the vector magnitude functor that we are creating.");
+  params.addParam<MooseFunctorName>(
+      "vector_functor", "The name of a vector functor that we will take the magnitude of.");
   return params;
 }
 
 template <bool is_ad>
 VectorMagnitudeFunctorMaterialTempl<is_ad>::VectorMagnitudeFunctorMaterialTempl(
     const InputParameters & parameters)
   : FunctorMaterial(parameters),
-    _x(getFunctor<GenericReal<is_ad>>("x_functor")),
+    _x(isParamValid("x_functor") ? &getFunctor<GenericReal<is_ad>>("x_functor") : nullptr),
     _y(getFunctor<GenericReal<is_ad>>("y_functor")),
-    _z(getFunctor<GenericReal<is_ad>>("z_functor"))
+    _z(getFunctor<GenericReal<is_ad>>("z_functor")),
+    _vector_functor(isParamValid("vector_functor")
+                        ? &getFunctor<VectorValue<GenericReal<is_ad>>>("vector_functor")
+                        : nullptr)
 {
+  if ((isParamValid("x_functor") + isParamValid("vector_functor")) != 1)
+    mooseError("Either a 'x_functor' (indicating the x-component) or 'vector_functor' parameter "
+               "must be provided to '",
+               name(),
+               "'");
+
```

Comment:
let s add a check for y_functor and z_functor not being provided by the user when vector_functor is

then good to go imo